test/functional/viml/max_args_spec.lua

```diff
@@ -0,0 +1,29 @@
+local helpers = require('test.functional.helpers')
+
+local clear = helpers.clear
+local eq = helpers.eq
+local exc_exec = helpers.exc_exec
+
+describe('Up to MAX_FUNC_ARGS arguments are handled by', function()
+  local max_func_args = 20  -- from eval.h
```

Comment:
This is fine. The way you can guard against a change to MAX_FUNC_ARGS  is to add a test that fails if more than `max_func_args` do _not_ cause an error.